pinot-common/src/main/java/com/linkedin/pinot/common/utils/FileUploadUtils.java

```diff
@@ -1,470 +1,349 @@
-import org.apache.http.entity.mime.FormBodyPart;
+import org.apache.http.entity.StringEntity;
 import org.apache.http.entity.mime.HttpMultipartMode;
-import org.apache.http.entity.mime.MultipartEntity;
+import org.apache.http.entity.mime.MultipartEntityBuilder;
+import org.apache.http.entity.mime.content.ContentBody;
+import org.apache.http.entity.mime.content.FileBody;
 import org.apache.http.entity.mime.content.InputStreamBody;
-import org.apache.http.impl.client.DefaultHttpClient;
-import org.apache.http.params.BasicHttpParams;
-import org.apache.http.params.CoreConnectionPNames;
-import org.apache.http.protocol.HTTP;
-import org.json.JSONObject;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClientBuilder;
 
 
 public class FileUploadUtils {
+  private FileUploadUtils() {
+  }
 
-  private static final Logger LOGGER = LoggerFactory.getLogger(FileUploadUtils.class);
-  public static final String HDR_CONTROLLER_HOST = "Pinot-Controller-Host";
-  public static final String HDR_CONTROLLER_VERSION = "Pinot-Controller-Version";
-  public static final String NOT_IN_RESPONSE = "NotInResponse";
-  private static final String SEGMENTS_PATH = "segments";
-  public static final String UPLOAD_TYPE = "UPLOAD_TYPE";
-  public static final String DOWNLOAD_URI = "DOWNLOAD_URI";
-  public static final int MAX_RETRIES = 5;
-  public static final int SLEEP_BETWEEN_RETRIES_IN_SECONDS = 60;
-  private static final MultiThreadedHttpConnectionManager CONNECTION_MANAGER =
-      new MultiThreadedHttpConnectionManager();
-  private static final HttpClient FILE_UPLOAD_HTTP_CLIENT = new HttpClient(CONNECTION_MANAGER);
-
-  {
-    FILE_UPLOAD_HTTP_CLIENT.getParams().setParameter("http.protocol.version", HttpVersion.HTTP_1_1);
-    FILE_UPLOAD_HTTP_CLIENT.getParams().setSoTimeout(3600 * 1000); // One hour
+  public static class CustomHeaders {
+    public static final String UPLOAD_TYPE = "UPLOAD_TYPE";
+    public static final String DOWNLOAD_URI = "DOWNLOAD_URI";
   }
 
-  public enum SendFileMethod {
-    POST {
-      public EntityEnclosingMethod forUri(String uri) {
-        return new PostMethod(uri);
-      }
-    },
-    PUT {
-      public EntityEnclosingMethod forUri(String uri) {
-        return new PutMethod(uri);
-      }
-    };
+  public static final int DEFAULT_SOCKET_TIMEOUT_MS = 3600 * 1000; // 1 hour
+
+  private static final CloseableHttpClient HTTP_CLIENT = HttpClientBuilder.create().build();
+  private static final String SCHEMA_PATH = "/schemas";
+  private static final String SEGMENT_PATH = "/segments";
 
-    public abstract EntityEnclosingMethod forUri(String uri);
+  public enum FileUploadType {
+    URI, JSON, TAR;
+
+    public static FileUploadType getDefaultUploadType() {
+      return TAR;
+    }
   }
 
-  public static int sendFile(final String host, final String port, final String path, final String fileName,
```

Comment:
Need to check if third-eye is using some of these APIs

Reply:
Good point, they do. Will update their code as well.